crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+    FraudProofVerificationInfoRequest, FraudProofVerificationInfoResponse,
+    InvalidDomainExtrinsicRootVerificationInfo,
+};
+use codec::Encode;
+use domain_block_preprocessor::runtime_api::InherentExtrinsicConstructor;
+use domain_block_preprocessor::runtime_api_light::RuntimeApiLight;
+use sp_api::{BlockT, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_core::traits::CodeExecutor;
+use sp_core::H256;
+use sp_domains::{DomainId, DomainsApi};
+use sp_runtime::traits::NumberFor;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+/// Trait to query and verify Domains Fraud proof.
+pub trait FraudProofHostFunctions: Send + Sync {
```

Comment:
This host function expects the verifier node to contain all the necessary data for verifying the fraud proof, it won't be held if we still allow customizing the `--state-pruning` and `--block-pruning` options for the non-archival consensus node, I remember @nazar-pc had mentioned the archiving process also have the similar requirement, we should make that requirement more explicit or plz leave a TODO if it is not going handles in this PR.

NOTE: `---block-pruning` is necessary since we need the block body to get the bundle.

Reply:
For obvious reasons we will not allow it, it is not really supported on Gemini 3f already.

Reply:
How was it approached? I still see the `--state-pruning` and `--block-pruning` options in the `gemini-3f-2023-oct-06` snapshot. Do we use a constant value for these options and ignore what the user pass in or make all consensus node to be archival nodes?

Reply:
Well, if you attempt to use it, things will break in a bad way. https://github.com/subspace/subspace/issues/1727 will address it by replacing Substrate CLI with our own CLI that will have more meaningful arguments (`--farmer` and `--operator` instead of `--validator`) and arguments that are not supported/not applicable will be removed.

Reply:
I see, but we may need the pruning dependent on the challenge period at least for nodes that turn on domain, in case archiving has a shorter requirement than the challenge period.

Reply:
Absolutely. Practically though it is not a concern yet, blocks are pruned so much later that it will not cause issues for now.